test/python/test_prefetched.py

```diff
@@ -0,0 +1,160 @@
+#  limitations under the License.
+# ==============================================================================
+"""nGraph TensorFlow bridge prefetch test
+
+"""
+import sys
+import pytest
+import getpass
+import tensorflow as tf
+from tensorflow.python.framework import dtypes
+tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
+
+import ngraph_bridge
+
+import numpy as np
+from common import NgraphTest
+
+import warnings
+warnings.filterwarnings('ignore', category=FutureWarning)
+
+import ngraph_bridge
+
+
+class TestPrefetched(NgraphTest):
+
+    def build_data_pipeline(self, input_array, map_function, batch_size):
+        dataset = (tf.data.Dataset.from_tensor_slices(
+            (tf.constant(input_array)
+            )).map(map_function).batch(batch_size).prefetch(1))
+
+        iterator = dataset.make_initializable_iterator()
+        data_to_be_prefetched_and_used = iterator.get_next()
+        return data_to_be_prefetched_and_used, iterator
+
+    def build_model1(self, input_array, c1, c2):
+        # Convert the numpy array to TF Tensor
+        input_f = tf.cast(input_array, tf.float32)
+
+        # Define the Ops
+        pl1 = tf.placeholder(dtype=dtypes.int32)
+        pl1_f = tf.cast(pl1, tf.float32)
+        pl2 = tf.placeholder(dtype=dtypes.int32)
+        pl2_f = tf.cast(pl2, tf.float32)
+
+        mul = tf.compat.v1.math.multiply(input_f, c1)
+        add = tf.compat.v1.math.add(mul, pl2_f)
+        add2 = add + pl1_f
+        output = add2 - c2
+        return output, pl1, pl2
+
+    def build_model2(self, input_array, c1, c2):
+        # Convert the numpy array to TF Tensor
+        input_f = tf.cast(input_array, tf.float32)
+
+        # Define the Ops
+        pl1 = tf.placeholder(dtype=dtypes.int32)
+        pl1_f = tf.cast(pl1, tf.float32)
+        pl2 = tf.placeholder(dtype=dtypes.int32)
+        pl2_f = tf.cast(pl2, tf.float32)
+
+        mul = tf.compat.v1.math.multiply(pl2_f, input_f)
+        add = tf.compat.v1.math.add(mul, c2)
+        add2 = add + pl1_f * c1
+        output = add2
+        return output, pl1, pl2
+
+    def __run_test(self, pipeline_creator, model):
+        # build model
+        input_array = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+        map_multiplier = 10
+        map_function = lambda x: x * map_multiplier
+        batch_size = 1
+        pipeline, iterator = pipeline_creator(input_array, map_function,
+                                              batch_size)
+
+        # some constants
+        c1 = 5.0
+        c2 = 10.0
+        model, pl1, pl2 = model(pipeline, c1, c2)
+
+        outputs = []
+
+        sess = tf.Session()
+
+        # Initialize the globals and the dataset
+        sess.run(iterator.initializer)
+
+        for i in range(1, 10):
+            output = sess.run(model, feed_dict={pl1: i, pl2: i + 3})
+            outputs.append(output)
+
+        return outputs
+
+    # test hangs when "NGRAPH_TF_DISABLE_DEASSIGN_CLUSTERS" is set
+    def test_prefetch1(self):
```

Comment:
Also, test_prefetch1 and test_prefetch2 are very similar. Maybe we can package it into a helper function and test_prefetch1 calls helper with build_model1 and test_prefetch2 calls helper with build_model2